test/content-services/categoriesApi.spec.ts

```diff
@@ -0,0 +1,237 @@
+import { expect } from 'chai';
+import { AlfrescoApiConfig } from '../../src/alfrescoApiConfig';
+import { AlfrescoApi } from '../../src/alfrescoApi';
+import { CategoriesApi } from '../../src/api/content-rest-api';
+import { EcmAuthMock } from '../../test/mockObjects';
+import { CategoriesMock } from '../mockObjects/content-services/categories.mock';
+import { CategoryPaging } from '../../src/api/content-rest-api/model/categoryPaging';
+import { CategoryEntry } from '../../src/api/content-rest-api/model/categoryEntry';
+
+describe('Categories', () => {
+    let authResponseMock: EcmAuthMock;
+    let categoriesMock: CategoriesMock;
+    let categoriesApi: CategoriesApi;
+
+    beforeEach((done) => {
+        const hostEcm = 'http://127.0.0.1:8080';
+
+        authResponseMock = new EcmAuthMock(hostEcm);
+        categoriesMock = new CategoriesMock();
+
+        authResponseMock.get201Response();
+        const alfrescoJsApi = new AlfrescoApi({
+            hostEcm
+        } as AlfrescoApiConfig);
+
+        alfrescoJsApi.login('admin', 'admin').then(() => done());
+        categoriesApi = new CategoriesApi(alfrescoJsApi);
+    });
+
+    it('get subcategories for category with categoryId should return 200 if all if ok', (done) => {
+        categoriesMock.get200ResponseSubcategories('-root-');
+        categoriesApi.getSubcategories('-root-').then((response: CategoryPaging) => {
+            expect(response.list.pagination.count).to.be.equal(2);
+            expect(response.list.entries[0].entry.parentId).to.be.equal('-root-');
+            expect(response.list.entries[0].entry.id).to.be.equal('testId1');
+            done();
+        });
+    });
+
+    it('get subcategories for not existing category should return 404', (done) => {
```

Comment:
Shouldn't all tests start with the word "should"?

Reply:
Changed